packages/integrations/mdx/src/index.ts

```diff
@@ -71,6 +71,7 @@ export default function mdx(mdxOptions: MdxOptions = {}): AstroIntegration {
 					// Note: disable `.md` (and other alternative extensions for markdown files like `.markdown`) support
 					format: 'mdx',
 					mdExtensions: [],
+					remarkRehypeOptions: config.markdown.remarkRehype,
```

Comment:
I'm noticing this applies `remarkRehype` _in all cases._ To stay consistent with `remark` and `rehype` plugin configs, we should only apply this when `mdxOptions.extendPlugins === 'markdown'`. I'd add a check for that here.

Reply:
I'd probably add `remarkRehype` to the `MdxOptions` as well, in keeping with how we mirror the markdown config for the MDX integration. A couple changes for this:
- Add `remarkRehype` to the `MdxOptions` type
- Spread `mdxOptions.remarkRehype` with `config.markdown.remarkRehype`, checking that `extendPlugins` option I noted above
- Add `remarkRehype` to the MDX README. Feel free to [link to our existing `remarkRehype` docs](https://docs.astro.build/en/reference/configuration-reference/#markdownremarkrehype) for simplicity.

A bit complex I know, and we _will_ simplify these APIs in the future! Just want to make sure we stay consistent with extending Markdown configs into MDX.

Reply:
@bholmesdev, thanks for the feedback! See latest commits for an attempt to take care of this.

Three things I'm unsure of:
- Is `remarkRehype` of type `RemarkRehype` or `RemarkRehypeOptions`? I would assume the latter, but the `markdown` options assume the former.
- Should `mdxOptions.remarkRehype` be extended with `config.markdown.remarkRehype` or the other way round?
- Is the README helpful enough?

Reply:
@bholmesdev, sounds good!

1. After finding https://github.com/mdx-js/mdx/blob/aff6de4fad5955b804d51cd109248425d786fb1c/packages/mdx/lib/core.js#L7 I have kept `RemarkRehypeOptions`.
2. Fixed and additional tests added.
3. Fixed.  Thanks a lot,  @sarah11918!